externallib/DPA/DP.py

```diff
@@ -0,0 +1,152 @@
+"""
+Module implementing the "Fast Search and Find of Density Peaks" clustering algorithm of Rodriguez and Laio (2014).
+"""
+
+import numpy as np
+import scipy as sp
+from scipy import stats
+from sklearn.neighbors import NearestNeighbors
+
+
+class Density_Peaks_clustering:
+    """
+    Clustering by fast search and find of density peaks, Rodriguez and Laio (2014).
+    """
+    def __init__(self, distances=None, indices=None, dens_type="eps", dc=None, percent=2.0):
```

Comment:
What is dens_type="eps"?

Reply:
When you estimate kernel density, there are two schemes:
1. you set a epsilon (a distance cutoff), and compute the density within this cutoff
or
2. you find k-NN, and compute the density using these k data points

Reply:
The only place I can find the **self.dens_type** instance attribute used is in the line:

`if self.dens_type == 'exp':`

Hence I don't understand the default value of "eps". I can't find it used anywhere.